### ISSUE TITLE
Pressing multiple times ENTER fast throws errors or results in strange behaviour

### ISSUE THREAD

**Issue by iseulde** [WordPress/gutenberg#2121]
Some things I noticed since #1989:

* After a pressing enter fast for a while, it stops creating new blocks.
* Sometimes it adds a line break in front of the block's content (also fast).
* Sometimes it throws this error:

```
Warning: validateDOMNesting(...): <p> cannot appear as a descendant of <p>.
```

Related: #2112

**Comment by hedgefield** [comment]
I don't seem to be able to reproduce the first point (how long would you say is "a while"?), but I do see the line breaks appearing sometimes - basically as if you pressed shift+enter in between enters.

Also, when backspacing all those new blocks, it seems the linebreaks 'travel up' too, combining into one block with multiple linebreaks along the way (or appending it to an existing block with text in it). It's like the way drops of water form a bigger puddle when they come in contact with each other, is the only way I can describe it.

**Comment by moorscode** [comment]
Holding the enter key will reproduce this for me as well.

**Comment by iseulde** [comment]
Should be fixed by ab44b9b91548bfb3cc360de5abc9d8a027ec1670 and 9784041d5af7fc4f489ff4547eaca0e16ec46dee.
